### PR TITLE
Update docs to include interface guidelines

### PR DESCRIPTION
**Pull request by danielguillan**
## Summary

This PR introduces an interim solution to provide design guidelines for Primer Brand components. It's based on the information architecture updates made in primer.style/design and uses the same new layout for navigating between "Overview" and "React" pages. 

With this updated we can add new design guidelines incrementally by:
1. Moving the current component mdx file into its own directory and renaming it (e.g., `Avatar.mdx` → `Avatar/react.mdx`)
2. Creating an `index.mdx` file for the design guidelines (e.g., `Avatar/index.mdx`)
3. Exporting the component layout from both files:
```javascript
import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout
```

## List of notable changes:

- Copied `base-layout`, `component-page-nav`, and `component-layout` from primer/design
- Adapted `component-layout` to support existing documentation
- Updated doctocat dependency to the latest version
- Avatar component updated to use the new layout
- Added initial Avatar design guidelines

_Note: We need to copy the layout and nav components from primer/design because these aren't available directly from Doctocat. Eventually, once the documentaion migration is completed, we can remove these altogether._

_Note 2: contrary to what we're doing in primer/design, we're not changing component page urls to use `kebab-case` yet, to avoid breaking existing links and keeping this PR focused._
## What should reviewers focus on?

- [Review the udpated Avatar doc pages](https://primer-89b744b258-26139705.drafts.github.io/components/Avatar)
- Ensure the updates doesn't break any other pages

## Supporting resources (related issues, external links, etc):

- https://github.com/github/primer/issues/2057

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

A screenshot showing the updated Avatar component page which now defaults to the Overview tab

A screenshot showing the React tab on the updated page for the Avatar component

### CODE REVIEW

File: apps/docs/src/layouts/component-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        <Heading as="h1">{title}</Heading>
+        {description ? (
+          <Text as="p" sx={{fontSize: 3, m: 0, mb: 3, maxWidth: '60ch'}}>
+            {description}
+          </Text>
+        ) : null}
+        <Box sx={{mb: 4}}>
+          <ComponentPageNav
+            basePath={basePath}
+            includeReact={reactId}
```
Comment (josepmartins) [inline apps/docs/src/layouts/component-layout.tsx:38]:
```suggestion
```

File: apps/docs/src/components/component-page-nav.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { UnderlineNav } from '@primer/react'
+import { Link as GatsbyLink } from 'gatsby'
+
+export function ComponentPageNav({
+  basePath,
+  current,
+}: {
+  basePath: string
+  includeReact?: boolean
```
Comment (josepmartins) [inline apps/docs/src/components/component-page-nav.tsx:10]: I think we can remove this prop for now.

```suggestion
```

**Review by josepmartins — APPROVED** [review]
LGTM! left two non-blocker comments

File: apps/docs/content/components/Avatar/index.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+### Size
+
+Avatar offers various sizes to fit different layout and visual scale needs. Sizes range from 32 to 80 pixels. Ensure the avatar is proportionate to its accompanying text.
+
+![An image showing the size scale of the avatar component: 32, 40, 48, 64, and 80 pixels](https://user-images.githubusercontent.com/175638/234826575-1d1d5019-f714-41b1-b8bc-c0d312a46cfe.png)
+
+### Shape
+
+Avatars appear as two different shapes: circle and square. The choice depends on the desired visual impression and the overall design aesthetic of your project.
```
Comment (rezrah) [inline apps/docs/content/components/Avatar/index.mdx:25]: Just checking that our Brand usage for avatar differs from product?

E.g. this is what the product avatar states:

- Circle avatars represent individual people.
- Square avatars represent non-human entities, such as bots, teams, or organizations

I recall for Galaxy recently, that we used square avatars for people, so assuming this is the case?

File: apps/docs/content/components/Avatar/index.mdx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Usage
+
+Use avatars to display a thumbnail that represents individuals in various contexts, such as testimonials, speaker sessions, blog post authors, and beyond.
+
+## Options
+
+### Size
+
+Avatar offers various sizes to fit different layout and visual scale needs. Sizes range from 32 to 80 pixels. Ensure the avatar is proportionate to its accompanying text.
```
Comment (rezrah) [inline apps/docs/content/components/Avatar/index.mdx:19]: Would be cool to see an example of a do/don't for this: 

> Ensure the avatar is proportionate to its accompanying text

These are great opportunities to show the example visually, to emphasise on the importance. Also just curious what a do/don't would look like here 😀

File: apps/docs/content/components/Avatar/index.mdx (hunk 3 above)
Comment (danielguillan) [reply apps/docs/content/components/Avatar/index.mdx:25]: Yes, it differs from product. We use both circle and square shaped avatars in Galaxy.

**Comment by danielguillan** [comment]
> My only feedback / slight concern would be that the React docs are getting buried a little now, and it seems like we're duplicating quite a lot content between overview and React.

Yes, that's expected in this very first iteration. Over time, all design guidelines will only live in the overview tab, while the react docs will be focused on the API and live code examples only, and auto-generated like Primer product react pages. Next step is to add the Figma docs too.

**Review by emilybrick — APPROVED** [review]

**Pull request merged**